types/block_test.go

```diff
@@ -906,3 +906,139 @@ func TestMessagesIsSorted(t *testing.T) {
 		})
 	}
 }
+
+func TestDataToProto(t *testing.T) {
+	type testCase struct {
+		name  string
+		input *Data
+		want  tmproto.Data
+	}
+	testCases := []testCase{
```

Comment:
does it make sense to test a `nil` case?  I know we are testing `empty` but `nil` would catch things never being initialized.

I would also add a two blob case as that would trigger a second iteration loop when applicable and could catch additional edge cases.

Reply:
resolved by https://github.com/celestiaorg/celestia-core/pull/890/commits/b7529b1e5ab35f34da5b88fdb91508685318c44d and https://github.com/celestiaorg/celestia-core/pull/890/commits/2f71ddc4adcda0795d18e204cd47ff310cd627b9